File: csharp/BookStore/src/BookStore/Book.cs
namespace BookStore
{
    public class Book : AbstractItem
    {
        private static readonly int DEFAULT_WEIGHT_BOOK_GRAMMS = 500;

        public Book(string name) : base(name)
        {
        }

        public override void PutIntoMyCart(Cart cart)
        {
            cart.Add(GetName());
        }

        protected override int ItemWeight()
        {
            if (GetName().Contains("C++"))
            {
                return 750;
            }
            else if (GetName().Contains("Good Parts"))
            {
                return 100;
            }
            return DEFAULT_WEIGHT_BOOK_GRAMMS;
        }
    }
}

File: cpp/src/Book.cpp
//
// Created by mille on 28-Sep-22.
//

#include "Book.h"

namespace bookstore {
  int Book::DEFAULT_WEIGHT_BOOK_GRAMMS = 500;

  Book::Book(std::string const& name) : AbstractItem(name) {
  }

  void Book::putIntoMyCart(Cart& cart) {
    cart.add(getName());
  }

  int Book::itemWeight() {
    if (getName().find("C++") != std::string::npos) {
      return 750;
    }
    else if (getName().find("Good Parts") != std::string::npos) {
      return 100;
    }
    return DEFAULT_WEIGHT_BOOK_GRAMMS;
  }
} // bookstore